### PR TITLE
MAIN B-21429 and B-21431

### PR DESCRIPTION
**Pull request by danieljordan-caci**
[INT PR 1](https://github.com/transcom/mymove/pull/13877)
[INT PR 2](https://github.com/transcom/mymove/pull/13898)

## Summary

We want to be able to track if a shipment is a domestic/international shipment, but we can only do that by checking the addresses under the shipment. If both addresses are CONUS then the `market_code` should be `d`, else it's gonna be `i`.

This does contain code needed for [B-21430](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21430) since the `isOconus` value is necessary to determine the shipment's market code.

This PR does the following:
- migration to populate `market_code` columns
- updates all Swagger definitions for shipments to send back `marketCode` to the UI
- updates all `model_to_payload.go` files to send back `marketCode`
- adds in logic in both shipment creation and update that determines if `marketCode` needs to change by checking the address values
- adds/updates relevant tests

### How to test

1. Fire up yer client & server
2. Check your migrations and the `mto_shipments` table - every value in `market_code` should say `d`
3. Create any shipment as a customer/office user with any CONUS addy
4. Throw in some console logs somewhere that you can see the data
5. You should see the `marketCode` value in the shipment object
6. Now... To test if a shipment changes to `i` you need to be able to enter an OCONUS state (in this case, Alaska)
7. Close down your server, in your `envrc` turn `FEATURE_FLAG_ENABLE_ALASKA` to `true`
8. Run `direnv allow` and fire up your server again
9. Create a shipment with an Alaska address
10. Check your logs, should now see `i` in the `marketCode` value on the shipment

## Screenshots
Screenshot 2024-10-10 at 1 29 48 PM

[Previous INT PR](https://github.com/transcom/mymove/pull/13877)

## Summary

An issue was discovered that when creating a PPM shipment an error was returning because the `eTag` was not correct.

In a previous implementation for populating the `market_code` value in the db, the flow was this:
- Create shipment
- Create PPM shipment
- Update shipment `market_code` based off of created address values in PPM shipment (a new `eTag`)

The new `eTag` was not being updated in the object sent back to the UI, so there was a difference in that value when progressing through the PPM flow. 

This PR corrects that by:
- updating `mtoShipment` object with queried `updated_at` value
- updating `mtoShipment` object with `market_code` IF it is different than the default `d`

### How to test

1. Access MM as a customer
2. Go through and create a PPM shipment
3. You should not receive any errors

### CODE REVIEW

File: pkg/services/address/address_creator_test.go [hunk 1]
```diff
@@ -30,6 +30,28 @@ func (suite *AddressSuite) TestAddressCreator() {
 		suite.Equal(city, address.City)
 		suite.Equal(state, address.State)
 		suite.Equal(postalCode, address.PostalCode)
+		suite.False(*address.IsOconus)
+		suite.Nil(address.StreetAddress2)
+		suite.NotNil(address.Country)
+	})
+
+	suite.Run("Successfully creates an OCONUS address with AK state", func() {
```
Comment (traskowskycaci) [inline pkg/services/address/address_creator_test.go:38]: Successfully creates an OCONUS address with AK state test is duplicated in this file

File: pkg/handlers/internalapi/internal/payloads/payloads_test.go
Comment (traskowskycaci) [inline pkg/handlers/internalapi/internal/payloads/payloads_test.go:1]: File not in INT PR but matches int branch, g2g

File: pkg/services/address/address_creator_test.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/services/address/address_creator_test.go]: fixeddddd

**Review by cameroncaci — APPROVED** [review]
commits match, but SHAs are differeeennntttttttttttttt - cherry picking agghh

code looks good

**Comment by danieljordan-caci** [comment]
> commits match, but SHAs are differeeennntttttttttttttt - cherry picking agghh
> 
> code looks good

Had to work from INT due to dependency work. Sorryyyyyy

**Review by TevinAdams — APPROVED** [review]
All commits from INT PRs are there. GTG

**Pull request merged**